lib/swiftype-app-search/exceptions.rb

```diff
@@ -16,9 +16,10 @@ class Forbidden < ClientException; end
   class InvalidDocument < ClientException; end
 
   class UnexpectedHTTPException < ClientException
-    def initialize(http_response)
-      @errors = []
-      super("HTTP #{http_response.code}: #{http_response.body}")
+    def initialize(response, response_body)
+      response_body['errors'] = [response.message] unless response_body['errors']
+      response_body['errors'].map! { |e| "(#{response.code}) #{e}" }
```

Comment:
We try to avoid mutating data inside collections, so I would rewrite this like
```
def initialize(response, response_json)
  errors = (response_json['errors'] || [response.message]).map { |e| "(#{response.code}) #{e}" }
  super({ 'errors' => errors })
end
```